tests/MongoDB.Driver.Tests/AggregateFluentTests.cs

```diff
@@ -1031,6 +1031,80 @@ public void ReplaceWith_should_add_the_expected_stage([Values(false, true)] bool
             }
         }
 
+        public class Item1
+        {
+            public ObjectId Id { get; set; }
+            public string Name1 { get; set; }
+        }
+
+        public class Item2
+        {
+            public ObjectId Id { get; set; }
+            public string Name2 { get; set; }
+        }
```

Comment:
I don't think converting Item2 to Item1 is a good example.

Consider instead converting both pipelines to this class:

    public class ItemResult
    {
        public string Name { get; set; }
    }

That will be the result of the union.

Reply:
done